ingest/scripts/parse-fields-from-mumps-strain.py

```diff
@@ -0,0 +1,115 @@
+        help="Strain field from which the division should be parsed.")
+    parser.add_argument("--division-field", default='division',
+        help="Division field to which the parsed division should be saved.")
+    parser.add_argument("--genotype-field", default='MuV_genotype',
+        help="MuV_genotype field to which the parsed genotype should be saved.")
+    parser.add_argument("--date-field", default='date',
+        help="Date field to which the parsed year should be saved.")
+
+    return parser.parse_args()
+
+def _parse_division_from_strain(record, strain_field):
+    strain = record.get(strain_field, '')
+    division_field = ''
+
+    # Expand pattern with list of country codes
+    pattern = rf"MuV[siS]/([^/0-9.]+)\.({'|'.join(COUNTRY_CODES)})/[0-9]{{1,2}}\.[0-9]{{2}}"
+    # Examples:
+    # Parse division 'Manitoba' from 'MuVs/Manitoba.CAN/17.17/5[G]'
+    # Parse division 'Ontario' from 'MuVi/Ontario.CAN/12.17/3[G]'
+    # Parse division 'Boras' from 'MuVs/Boras.SWE/2.20[G]'
+    # Avoid parsing: '14778' from 'MuVs/14778.SWE/0.06[G]
+    if re.match(pattern, strain):
+        match = re.match(pattern, strain)
+        division_field = match.group(1)
+
+    # Parse "NewYork" from 'MuVs/NewYork.USA/2019/38854'
+    # MuVs/NewYork.USA/2017/51025
+    if re.match(r'MuV[siS]/([^/0-9.]+)\.USA/', strain):
+        match = re.match(r'MuV[siS]/([^/0-9.]+)\.USA/', strain)
+        division_field = match.group(1)
+
+    # Parse "Indiana" from 'MuVs/Indiana_USA/201738605'
+    if re.match(r'MuV[siS]/([^/0-9.]+)_USA/', strain):
+        match = re.match(r'MuV[siS]/([^/0-9.]+)_USA/', strain)
+        division_field = match.group(1)
+
+    # Parse "Okinawa" from "MuVi/Okinawa50.JPN/29.15[G]"
+    if re.match(r'MuV[siS]/([^/0-9.]+)[0-9]+\.JPN/', strain):
+        match = re.match(r'MuV[siS]/([^/0-9.]+)[0-9]+\.JPN/', strain)
+        division_field = match.group(1)
+
+    return division_field
+
+def _parse_genotype_from_strain(record, strain_field):
+    strain = record.get(strain_field, '')
+    genotype_field = ''
+
+    if re.search(r'\[([A-Z])\]$', strain):
+        match = re.search(r'\[([A-Z])\]$', strain)
+        genotype_field = match.group(1)
+
+    return genotype_field
+
+def _parse_date_from_strain(record, strain_field):
+    strain = record.get(strain_field, '')
+    date_field = 'XXXX-XX-XX'
+
+    # Expand pattern with list of country codes
+    pattern = rf"MuV[siS]/[^/]+\.({'|'.join(COUNTRY_CODES)})/[0-9]{{1,2}}\.([0-9]{{2}})"
+    # Examples:
+    # Parse year '2017' from 'MuVs/Manitoba.CAN/17.17/5[G]'
+    # Parse year '2006' from 'MuVs/14778.SWE/0.06[G]
+    if re.match(pattern, strain):
+        match = re.match(pattern, strain)
+        year_suffix = int(match.group(2))
+        # TODO: The 30 threshhold will need to be revisited in year 2030
+        if year_suffix > 30:
+            date_field = f"19{year_suffix:02d}-XX-XX"
+        else:
+            date_field = f"20{year_suffix:02d}-XX-XX"
```

Comment:
Is there a particular submitter that uses this 2-digit year pattern?

Reply:
Good question! The 2-digit year pattern seems to be a standard convention for mumps, and is mostly followed, based on a quick perusal of the [metadata.tsv](https://data.nextstrain.org/files/workflows/mumps/metadata.tsv.zst), with a few exceptions. Out of 14,547 mumps records, ~12,726 entries seem to have the 2-digit year format, while perhaps ~41 have the 4-digit year pattern.

The first set of numbers is epiweeks while the second set of numbers is the 2-digit year  (e.g. `MuVi/Yamagata.JPN/12.12[G1]` -> epiweek=12, year = 2012). I haven't attempted to parse the epiweek to a particular month since months can start mid-week.

Good to ask! Personally I prefer the 4-digit year precisely to avoid this kind of work-around.

Reply:
> The first set of numbers is epiweeks while the second set of numbers is the 2-digit year

I think this may be a WHO standard; rubella strains are named in a similar way, with an epiweek/2 digit year format.